dev/releases/utils.py

```diff
@@ -143,7 +143,7 @@ def check_whether_git_tag_exists(tag):
     return False
 
 # Returns a boolean
-def check_whether_github_release_exists(tag):
+def does_github_release_exist(tag):
     if CURRENT_REPO == None:
```

Comment:
Why is `CURRENT_REPO` even a global variable? In general I would try to avoid such things, and instead let `initialize_github` return the repo object, then have the calling code store it in a local var, say `repo` and then pass it around as an argument. And when it comes down to this, this function is equivalent to `any(r.tag_name == tag for r in repo.get_releases())` so it might not even be necessary to have this as a dedicated function.

Reply:
> `any(r.tag_name == tag for r in repo.get_releases())`

Ooh, I didn't know that syntax. That's really nice. Yep, I'm getting rid of this function.

Wrt the `CURRENT_REPO`: I think we won't be changing repos in the middle of the script, so I've decided to use `CURRENT_REPO` as a global variable which we have to initialize once at the beginning of our script and then never change again. But I'm not married to that. If you want I can open a new PR and remove the global variable `CURRENT_REPO`. That should be easy.

Reply:
Ah wait, it's just list comprehension inside an `any()` so you can drop the square brackets? :thinking: Anyways, I'm using it!

Reply:
And I got rid of that function completely. Nice.

Reply:
I just dislike global state, it makes it generally harder to reason about the program, and here it doesn't seem to add anything useful?

Reply:
It also is the sole reason why you can't just do `import * from utils` as in all other of our files

Reply:
> It also is the sole reason why you can't just do import * from utils as in all other of our files

Yeah, I wasn't aware of this until I tried it out. :/